### PR TITLE
update depot_tools to cci.20201009

### PR DESCRIPTION
**Pull request by boussaffawalid**
Specify library name and version:  **lib/1.0**

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

Same as https://github.com/conan-io/conan-center-index/pull/3182 but included a patch for windows
ping @reneme 

closes #3182

### CODE REVIEW

File: recipes/depot_tools/all/conandata.yml [hunk 1]
```diff
@@ -1,3 +1,10 @@
 sources:
   "20200407":
     url: "https://chromium.googlesource.com/chromium/tools/depot_tools/+archive/48c5c9c50455c625c58dab2fbb0904cac467168c.tar.gz"
+  "cci.20201009":
+    url: "https://chromium.googlesource.com/chromium/tools/depot_tools.git/+archive/b073999c6f90103a36a923e63ae8cf7a5c9c6c8c.tar.gz"
+patches:
+  "20200407": []
```
Comment (madebr) [inline recipes/depot_tools/all/conandata.yml:7]:
```suggestion
```

File: recipes/depot_tools/all/conanfile.py [hunk 2]
```diff
@@ -40,6 +42,9 @@ def source(self):
         tools.get(**self.conan_data["sources"][self.version], destination=self._source_subfolder)
         self._dereference_symlinks()
 
+        for patch in self.conan_data["patches"][self.version]:
+            tools.patch(**patch)
```
Comment (madebr) [inline recipes/depot_tools/all/conanfile.py:46]:
```suggestion
        for patch in self.conan_data.get("patches", {}).get(self.version, []):
            tools.patch(**patch)
```

**Comment by prince-chrismc** [comment]
please add `closes #3182` in the top comment so we won't have duplicates

**Review by prince-chrismc — APPROVED** [review]
Since this has no SHA is it worth our while to patch in build?

**Review by SSE4 — APPROVED** [review]

**Comment by reneme** [comment]
Nice, thanks! Is that enough to make the recipe work including `DEPOT_TOOLS_UPDATE=0`?
Also: I didn't choose the 20201009 upstream version for any particular reason. Maybe we could update to a more recent revision of depot tools instead?

**Comment by boussaffawalid** [comment]
Actually this still does not work. it fails again on windows when running `gclient sync`, because git.bat cannot be found which is created when DEPOT_TOOLS_UPDATE=1.

As a work around I created a script git.bat in the package bin folder which call system git, that works! 

Now, I tried to build crashpad, ironically it fail because crashpad needs both python2 and python3.

I see three options here:
* patch both vpython3.bat and vpython2.bat but this needs both python versions to be installed, or two conan packages for python 2 and 3
* change the behavior of DEPOT_TOOLS_UPDATE=1 to only fetch python2, python3 and git.
* wait until this get fixed upstream

**Review by Croydon — APPROVED** [review]

**Pull request merged**